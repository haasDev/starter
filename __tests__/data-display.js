import React from 'react';
import DataDisplay from '../src/data-display';
import {mount} from 'enzyme';
import snapshot from './snapshot';

describe('DataDisplay', () => {
  const projectMap = {
    1: {id: 1, name: 'Foo'},
    2: {id: 2, name: 'Bar'}
  };

  it('should render', () => {
    snapshot(<DataDisplay projectMap={projectMap} />);
  });

  it('should delete a project', () => {
    // Mock the fetch call to delete a project.
    window.BASE_URL = 'https://localhost';
    window.fetch = jest.fn().mockImplementation((url, options) => {
      expect(url).toContain(`${window.BASE_URL}/project/`);
      expect(options.method).toBe('DELETE');
      return Promise.resolve({ok: true});
    });

    // Mock the React setState method that the app makes global.
    window.setState = fn => {
      const state = {projectMap};
      const newState = fn(state);
      expect(newState).toEqual({projectMap: {1: projectMap[1]}});
      return newState;
    };
    const wrapper = mount(<DataDisplay projectMap={projectMap} />);

    // Find the delete button for "bar".
    // It is the first one because "bar" precedes "foo" alphabetically.
    const deleteBtn = wrapper.find('.delete-btn').first();

    deleteBtn.simulate('click');
  });
});
