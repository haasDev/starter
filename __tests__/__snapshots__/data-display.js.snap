exports[`DataDisplay should render 1`] = `
<div
  className="data-display">
  <table>
    <caption>
      Projects
    </caption>
    <thead>
      <tr>
        <th />
        <th>
          Name
        </th>
        <th>
          Description
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <button
            className="delete-btn"
            onClick={[Function]}>
            ✖
          </button>
        </td>
        <td>
          Bar
        </td>
        <td />
      </tr>
      <tr>
        <td>
          <button
            className="delete-btn"
            onClick={[Function]}>
            ✖
          </button>
        </td>
        <td>
          Foo
        </td>
        <td />
      </tr>
    </tbody>
  </table>
  <a
    href="#entry">
    Add Projects
  </a>
</div>
`;
